src/analysis/bits-bounds-lattice.h

```diff
@@ -0,0 +1,144 @@
+#ifndef wasm_analysis_bits_bounds_lattice_h
+#define wasm_analysis_bits_bounds_lattice_h
+
+#include <optional>
+
+#include "lattice.h"
+#include "wasm.h"
+
+namespace wasm::analysis {
+
+struct MaxBitsLattice {
+  enum LatticeState { BOTTOM, REGULAR_VALUE, TOP };
+
+  struct Element {
+    Index upperBound = 0;
```

Comment:
We discussed various options here. From the name, I can't tell which one ended up used? Seems like a good place for a comment.

I also don't have a good guess about what "constVal" is right below it.

Reply:
Added a comment explaining the values now. 

The two options we discussed are using a max bits range (and potentially a min bits range in the future), or using a literal value all the time. For cases where we don't know the literal value, we could set all bits right of the max bit to 1. We can "or" these literal bits together when joining or increasing the max bits count.

This implementation follows the max bits range scheme mostly. I added an option to put a literal as a value instead of a max bits range in the case of constants. This is because in `bits.h`, some of the calculations involve operations directly on constant values.

However, I think some of these operations like the ceil log 2 can probably be replaced with a max bits bound, so maybe we can even get rid of the literal value.